tests/test_model_bom.py

```diff
@@ -139,6 +139,15 @@ def test_empty_bom(self) -> None:
         self.assertFalse(bom.services)
         self.assertFalse(bom.external_references)
 
+    def test_root_component_only_bom(self) -> None:
+        with warnings.catch_warnings():
```

Comment:
please use python `unittest` framework mechanisms: https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertWarns

Reply:
I only found assertWarns, but I need assertNotWarns which seems not to exist: https://discuss.python.org/t/add-assertnotwarns-method-to-unittest-testcase/27351
The action parameter in catch_warning was only added in Python 3.11: https://docs.python.org/3/library/warnings.html#warnings.catch_warnings
How should I proceed here?